source/fundamentals/serialization/poco.txt

```diff
@@ -261,6 +261,167 @@ The following code sample maps the ``Identifier`` property to the
+{+driver-short+}:
+
+.. list-table::
+   :header-rows: 1
+   :widths: 25 75
+
+   * - Id Field Data Type
+     - How to Use
+
+   * - ``Guid``
+     - To use the COMB algorithm to generate a unique ``Guid`` value, apply the
+       ``[BsonId(IdGenerator = typeof(CombGuidGenerator))]`` attribute to the ``Id``
+       property, as shown in the following example:
+
+       .. code-block:: csharp
+          :copyable: true
+
+          public class House
+          {
+             [BsonId(IdGenerator = typeof(CombGuidGenerator))]
+             public Guid Id { get; set; }
+          }
+
+       To generate a unique ``Guid`` value without the COMB algorithm, don't apply
+       the preceding attribute to the ``Id`` property. The driver automatically uses
+       the ``GuidGenerator`` type to generate a unique value for the ``Id`` field.
+
+       .. code-block:: csharp
+          :copyable: true
+
+          public class House
+          {
+             public Guid Id { get; set; }
+          } 
+
+   * - ``ObjectId``
+     - The driver automatically uses the ``ObjectIdGenerator`` type for ``Id`` fields with
+       the ``ObjectId`` data type, such as the one in the following code example:
+
+       .. code-block:: csharp
+          :copyable: true
+
+          public class House
+          {
+             public ObjectId Id { get; set; }
+          } 
+
+   * - ``string``
+     - To serialize the ``Id`` property as a ``string``, apply the
+       ``[BsonID(IdGenerator = typeof(StringObjectIdGenerator))]``
+       attribute to the property, as shown in the following code example. The driver
+       uses the ``StringObjectIdGenerator`` type to generate a unique ``ObjectId`` value,
+       convert it to a ``string``, and assign it to the ``Id`` property.
+
+       .. code-block:: csharp
+          :copyable: true
+   
+          public class House
+          {
+             [BsonId(IdGenerator = typeof(StringObjectIdGenerator))]
+             public string Id { get; set; }
+          }
+         
+       To serialize the ``Id`` property as an ``ObjectId``, apply the
+       ``[BsonRepresentation(BsonType.ObjectId)]`` attribute to the property, as shown
+         in the following code example. In this case, if the property contains the value
+         ``null``, the driver uses the ``ObjectIdGenerator`` type to generate a unique
+         ``ObjectId`` value and assigns it to the ``Id`` property.
+
+         .. code-block:: csharp
+            :copyable: true
+   
+            public class House
+            {
+               [BsonRepresentation(BsonType.ObjectId)]
+               public string Id { get; set; }
+            }
+       
+       If you apply both the ``[BsonId(IdGenerator = typeof(StringObjectIdGenerator))]``
+       and ``[BsonRepresentation(BsonType.ObjectId)]`` attributes to the ``Id`` property,
+       the driver generates an ``ObjectId`` value for the field.
```

Comment:
Not sure this is worded the best way.

If you apply both you have simply done TWO different things:

1. Configured the representation of the ID in the database
2. Configured the id generator to use for missing ID values